packages/cli/src/util/link/handle-git-connection.ts

```diff
@@ -0,0 +1,271 @@
+import { Dictionary } from '@vercel/client';
+import {
+  connectGitProvider,
+  disconnectGitProvider,
+  formatProvider,
+  parseRepoUrl,
+} from '../git/connect-git-provider';
+import { Output } from '../output';
+import list from '../input/list';
+import Client from '../client';
+import { Org, Project, ProjectSettings } from '../../types';
+import { getCommandName } from '../pkg-name';
+import updateProject from '../projects/update-project';
+import chalk from 'chalk';
+
+function getProjectSettings(project: Project): ProjectSettings {
```

Comment:
Is this function necessary? Couldn't you just do `settings || project` below instead?

Reply:
I removed the function, but brought it back briefly. `settings || project` doesn't work because the API expects a list of project settings, so additional `Project` properties like `accountId` make the request fail. I initially created this function because I needed a way to strip the extra properties from `project` to turn it into the type `ProjectSettings`.

I'm sure there's a better way to do this, though.